dbptk-modules/dbptk-filter-inventory/src/main/java/com/databasepreservation/modules/inventory/InventoryFilter.java

```diff
@@ -0,0 +1,209 @@
+  private char separator;
+
+  private DatabaseFilterModule exportModule;
+  private DatabaseStructure databaseStructure;
+
+  private String currentSchemaName;
+  private String currentTableName;
+  private Map<Integer,String> listOfHeaders;
+  private List<Map> listOfData;
+
+  private File csv_file;
+  private File csv_dir_path;
+  private OutputStreamWriter outStreamWriter;
+  private CSVPrinter csv_printer;
+
+  public InventoryFilter(String prefix, String dirPath, boolean pPrintHeader, char separator) {
+    this.prefixPath = prefix;
+    this.dirPath = dirPath;
+    this.printHeader = pPrintHeader;
+    this.separator = separator;
+  }
+
+  @Override
+  public void initDatabase() throws ModuleException {
+    this.exportModule.initDatabase();
+  }
+
+  @Override
+  public void setIgnoredSchemas(Set<String> ignoredSchemas) {
+    this.exportModule.setIgnoredSchemas(ignoredSchemas);
+  }
+
+  @Override
+  public void handleStructure(DatabaseStructure structure) throws ModuleException {
+    this.databaseStructure = structure;
+    this.exportModule.handleStructure(structure);
+  }
+
+  @Override
+  public void handleDataOpenSchema(String schemaName) throws ModuleException {
+    this.currentSchemaName=schemaName;
+    if(!this.dirPath.equals("")){
+      this.csv_dir_path = new File(this.dirPath);
+      this.csv_dir_path.mkdirs();
+    }
+
+    this.exportModule.handleDataOpenSchema(schemaName);
+  }
+
+  @Override
+  public void handleDataOpenTable(String tableId) throws ModuleException {
+    TableStructure currentTable = databaseStructure.getTableById(tableId);
+    this.currentTableName = currentTable.getId();
+    if(!this.dirPath.equals("") && !this.prefixPath.equals("")) {
+      this.csv_file = new File(this.csv_dir_path, this.prefixPath + "-" + this.currentTableName + ".csv");
+    }else if(!this.dirPath.equals("") && this.prefixPath.equals("")){
+      this.csv_file = new File(this.csv_dir_path,  this.currentTableName + ".csv");
+    }else if(this.dirPath.equals("") && !this.prefixPath.equals("")){
+      this.csv_file = new File(  this.prefixPath + "-" + this.currentTableName + ".csv");
+    } else{
+      this.csv_file = new File( this.currentTableName + ".csv");
+    }
+
+    try {
+      this.csv_file.createNewFile();
+      this.outStreamWriter = new FileWriter(csv_file);
+      this.csv_printer = CSVFormat.newFormat(this.separator).withRecordSeparator("\n").print(this.outStreamWriter);
+
+      this.listOfHeaders = new HashMap<>();
+      this.listOfData = new ArrayList<>();
+
+      int index = 0;
+
+      for (ColumnStructure column : currentTable.getColumns()) {
+        if (currentTable.isFromCustomView() || ModuleConfigurationManager.getInstance().getModuleConfiguration().isInventoryColumn(currentTable.getSchema(),
+                currentTable.getName(), column.getName())) {
+          String header = this.currentTableName + "." + column.getName();
+          this.listOfHeaders.put(index,header);
```

Comment:
change to List data type since the header is being printed below in this method anyways